File: CompilationTest/BigInteger/results/BigInteger.java
import java.math.BigInteger;

class MathUtils {
    public static Integer calc(Integer n) throws Exception
    {
        Integer result = 1;
        for (Integer i = 2; i <= n; i++) {
            result = result * i;
            if (result > 10) {
                result = result >> 2;
            }
        }
        return result;
    }
    
    public static BigInteger calcBig(Integer n) throws Exception
    {
        BigInteger result = BigInteger.valueOf(1);
        for (Integer i = 2; i <= n; i++) {
            result = result.multiply(BigInteger.valueOf(i)).add(BigInteger.valueOf(123));
            result = result.add(result);
            if (result.compareTo(BigInteger.valueOf(10)) > 0) {
                result = result.shiftRight(2);
            }
        }
        return result;
    }
}

class Program {
    public static void main(String[] args) throws Exception {
        System.out.println("5 -> " + MathUtils.calc(5) + ", 24 -> " + MathUtils.calcBig(24));
    }
}

File: CompilationTest/BigInteger/results/BigInteger.cpp
#include <OneLang-Core-v0.1/one.hpp>
#include <iostream>

class MathUtils {
  public:
    static int calc(int n) {
        int result = 1;
        for (int i = 2; i <= n; i++) {
            result = result * i;
            if (result > 10) {
                result = result >> 2;
            }
        }
        return result;
    }
    
    static one::sp<OneBigInteger> calcBig(int n) {
        auto result = OneBigInteger::fromInt(1);
        for (int i = 2; i <= n; i++) {
            result = result * i + 123;
            result = result + result;
            if (result > 10) {
                result = result >> 2;
            }
        }
        return result;
    }

  private:
};

int main()
{
    std::cout << std::string("5 -> ") + std::to_string(MathUtils::calc(5)) + ", 24 -> " + MathUtils::calcBig(24) << std::endl;
    return 0;
}